src/cbor.c

```diff
@@ -405,6 +405,30 @@ int cbor_deserialize_int16(const uint8_t *data, int16_t *value)
     return 0;
 }
 
+int cbor_deserialize_uint8(const uint8_t *data, uint8_t *value)
+{
+    uint32_t tmp;
+    int size = cbor_deserialize_uint32(data, &tmp); // also checks value for null-pointer
+
+    if (size > 0 && tmp <= UINT8_MAX) {
+        *value = tmp;
+        return size;
+    }
+    return 0;
+}
+
+int cbor_deserialize_int8(const uint8_t *data, int8_t *value)
+{
+    int32_t tmp;
+    int size = cbor_deserialize_int32(data, &tmp); // also checks value for null-pointer
+
+    if (size > 0 && tmp <= INT8_MAX && tmp >= INT8_MIN) {
```

Comment:
It should be worth noting in the docs that any out-of-range value becomes 0, silently without any error being logged. We could perhaps open a feature request in future that would log or flag these errors somehow, as they are quite common with data being squashed to 8 bits, and can be difficult to debug.

Reply:
Yeah, and it is actually checked. The 0 is propagated to below line:
https://github.com/ThingSet/thingset-device-library/blob/1a831a93717acb0a4740f41b69e1089a51b1a027/src/thingset_bin.c#L400

which is finally checked here and should return "bad request" in case of 0:
https://github.com/ThingSet/thingset-device-library/blob/1a831a93717acb0a4740f41b69e1089a51b1a027/src/thingset_bin.c#L417-L419